docs/source/1.0/spec/aws/aws-cloudformation.rst

```diff
@@ -0,0 +1,672 @@
+        }
+
+        structure CreateFooRequest {
+            @mutableProperty
+            tags: TagList,
+        }
+
+        structure CreateFooResponse {
+            fooId: String,
+        }
+
+        structure FooProperties {
+            @mutableProperty
+            barProperty: String,
+        }
+
+
+.. _aws.cloudformation#createOnlyProperty-trait:
+
+``aws.cloudformation#createOnlyProperty`` trait
+-----------------------------------------------
+
+Summary
+    Indicates that the CloudFormation property generated from this member can
+    be specified only during resource creation and can be returned in a `read`
+    or `list` request.
+Trait selector
+    ``structure > member``
+
+    *Any structure member*
+Value type
+    Annotation trait
+Conflicts with
+    :ref:`aws.cloudformation#additionalIdentifier-trait`,
+    :ref:`aws.cloudformation#excludeProperty-trait`,
+    :ref:`aws.cloudformation#mutableProperty-trait`,
+    :ref:`aws.cloudformation#readOnlyProperty-trait`,
+    :ref:`aws.cloudformation#writeOnlyProperty-trait`
+
+The following example defines a CloudFormation resource that has the derivable
+``immutableSetting`` property marked as create only:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#createOnlyProperty
+        use aws.cloudformation#resource
+
+        @resource(additionalSchemas: FooProperties)
+        resource Foo {
+            identifiers: {
+                fooId: String,
+            },
+        }
+
+        structure FooProperties {
+            @createOnlyProperty
+            immutableSetting: Boolean,
+        }
+
+
+.. _aws.cloudformation#readOnlyProperty-trait:
+
+``aws.cloudformation#readOnlyProperty`` trait
+---------------------------------------------
+
+Summary
+    Indicates that the CloudFormation property generated from this member can
+    be returned by a `read` or `list` request, but cannot be set by the user.
+Trait selector
+    ``structure > member``
+
+    *Any structure member*
+Value type
+    Annotation trait
+Conflicts with
+    :ref:`aws.cloudformation#createOnlyProperty-trait`,
+    :ref:`aws.cloudformation#excludeProperty-trait`,
+    :ref:`aws.cloudformation#mutableProperty-trait`,
+    :ref:`aws.cloudformation#writeOnlyProperty-trait`
+
+The following example defines a CloudFormation resource that has the derivable
+``updatedAt`` and ``createdAt`` properties marked as read only:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#readOnlyProperty
+        use aws.cloudformation#resource
+
+        @resource(additionalSchemas: FooProperties)
+        resource Foo {
+            identifiers: {
+                fooId: String,
+            },
+            get: GetFoo,
+        }
+
+        operation GetFoo {
+            input: GetFooRequest,
+            output: GetFooResponse,
+        }
+
+        structure GetFooRequest {
+            fooId: String
```

Comment:
This needs to be required. I suggest running all these examples through validation.